File: dotnet/backend/Models/booking_header.cs
using System;
using System.Collections.Generic;

namespace Etour_Backend_dotnet.Models;

public partial class booking_header
{
    public int booking_id { get; set; }

    public DateOnly booking_date { get; set; }

    public int customer_id { get; set; }

    public int tour_id { get; set; }

    public int departure_date_id { get; set; }

    public int total_passengers { get; set; }

    public decimal tour_amount { get; set; }

    public decimal tax_amount { get; set; }

    public decimal total_amount { get; set; }

    public string booking_status { get; set; } = null!;

    public virtual ICollection<booking_addon_master> booking_addon_master { get; set; } = new List<booking_addon_master>();

    public virtual customer_master customer { get; set; } = null!;

    public virtual departure_date_master departure_date { get; set; } = null!;

    public virtual ICollection<passenger_master> passenger_master { get; set; } = new List<passenger_master>();

    public virtual ICollection<payment> payment { get; set; } = new List<payment>();

    public virtual tour_master tour { get; set; } = null!;
}

File: java/backend/src/main/java/com/etour/app/entity/BookingHeader.java
package com.etour.app.entity;

import jakarta.persistence.*;

import java.math.BigDecimal;
import java.time.LocalDate;

@Entity
@Table(name = "booking_header")
public class BookingHeader {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "booking_id", nullable = false)
    private Integer id;

    @Column(name = "booking_date", nullable = false)
    private LocalDate bookingDate;

    @ManyToOne(fetch = FetchType.LAZY, optional = false)
    @JoinColumn(name = "customer_id", nullable = false)
    private CustomerMaster customer;

    @ManyToOne(fetch = FetchType.LAZY, optional = false)
    @JoinColumn(name = "tour_id", nullable = false)
    private TourMaster tour;

    @ManyToOne(fetch = FetchType.LAZY, optional = false)
    @JoinColumn(name = "departure_date_id", nullable = false)
    private DepartureDateMaster departureDate;

    @Column(name = "total_passengers", nullable = false)
    private Integer totalPassengers;

    @Column(name = "tour_amount", nullable = false, precision = 10, scale = 2)
    private BigDecimal tourAmount;

    @Column(name = "tax_amount", nullable = false, precision = 10, scale = 2)
    private BigDecimal taxAmount;

    @Column(name = "total_amount", nullable = false, precision = 10, scale = 2)
    private BigDecimal totalAmount;

    @Column(name = "booking_status", nullable = false, length = 20)
    private String bookingStatus = "PENDING";

    public Integer getId() {
        return id;
    }

    public void setId(Integer id) {
        this.id = id;
    }

    public LocalDate getBookingDate() {
        return bookingDate;
    }

    public void setBookingDate(LocalDate bookingDate) {
        this.bookingDate = bookingDate;
    }

    public CustomerMaster getCustomer() {
        return customer;
    }

    public void setCustomer(CustomerMaster customer) {
        this.customer = customer;
    }

    public TourMaster getTour() {
        return tour;
    }

    public void setTour(TourMaster tour) {
        this.tour = tour;
    }

    public DepartureDateMaster getDepartureDate() {
        return departureDate;
    }

    public void setDepartureDate(DepartureDateMaster departureDate) {
        this.departureDate = departureDate;
    }

    public Integer getTotalPassengers() {
        return totalPassengers;
    }

    public void setTotalPassengers(Integer totalPassengers) {
        this.totalPassengers = totalPassengers;
    }

    public BigDecimal getTourAmount() {
        return tourAmount;
    }

    public void setTourAmount(BigDecimal tourAmount) {
        this.tourAmount = tourAmount;
    }

    public BigDecimal getTaxAmount() {
        return taxAmount;
    }

    public void setTaxAmount(BigDecimal taxAmount) {
        this.taxAmount = taxAmount;
    }

    public BigDecimal getTotalAmount() {
        return totalAmount;
    }

    public void setTotalAmount(BigDecimal totalAmount) {
        this.totalAmount = totalAmount;
    }

    public String getBookingStatus() {
        return bookingStatus;
    }

    public void setBookingStatus(String bookingStatus) {
        this.bookingStatus = bookingStatus;
    }

}